File: C++/739. Daily Temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        int n = temperatures.size(); 
        vector<int> res(n); 
        stack<int> stk; 
        for (int i = 0; i < n; ++i) {
            while (stk.size() && temperatures[stk.top()] < temperatures[i]) {
                int ii = stk.top(); stk.pop(); 
                res[ii] = i - ii; 
            }
            stk.push(i); 
        }
        return res; 
    }
};

File: JAVA/739. Daily Temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temp)
    {
        Stack <Integer> list =new Stack<>();
        int l=temp.length;
        int ans[]=new int[l];
        int ind=l-1;
        int top=0;
        list.push(l-1);
        ans[ind]=0;
        ind--;
        for(int i=l-2;i>=0;i--)
        {
            if(temp[i]<temp[list.get(top)])
            {
                ans[ind]=list.get(top)-i;
                ind--;
                list.push(i);
                top+=1;
            }
            else
            {
                while (top>=0 && temp[i]>=temp[list.get(top)])
                {
                    top-=1;
                    list.pop();
                }
                if(top<0)
                    ans[ind]=0;
                else
                    ans[ind]=list.get(top)-i;
                ind--;
                list.push(i);
                top+=1;
            }
        }
        
        return ans;
        
    }
}
